indexer/pkg/common/status.go

```diff
@@ -39,10 +38,6 @@ func (m MessageStatus) String() string {
 	}
 }
 
-func (m MessageStatus) MarshalJSON() ([]byte, error) {
-	return json.Marshal(m.String())
-}
```

Comment:
There were so many copies of this type that it's not clear to me where this was used. In any case, overriding MarshalJSON this way means encoding is not symmetrical. I have opted to remove the special encoding rather than introduce JSON-Only support with `UnmarshalJSON`.